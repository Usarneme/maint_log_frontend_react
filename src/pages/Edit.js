import React from 'react'
import { Redirect, useParams } from 'react-router-dom'

import LogForm from '../components/LogForm'

function Edit(props) {
  const { id } = useParams()
  const isLoggedIn = (props.user && props.user.cookies ? props.user.cookies.length > 0 : false)
  if (!isLoggedIn) return <Redirect to="/welcome" />

  // do not display while data is being fetched/loaded/mounted
  // TODO make this part of an error boundary
  if (!props || !props.user || !props.user.log) return null
  const log = props.user.log.filter(entry => entry.id === id)
  // console.log('Edit page. Pushing log '+id+' to LogForm component...')
  // console.dir(log[0])

  return <LogForm log={log[0]} {...props} />
}

export default Edit
