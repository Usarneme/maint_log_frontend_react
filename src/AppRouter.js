import React from 'react'
import {
  BrowserRouter as Router,
  Switch,
  Route
} from "react-router-dom"

import Nav from './components/Nav'

import Settings from './pages/Settings'
import Add from './pages/Add'
import Edit from './pages/Edit'
import GuestHome from './pages/GuestHome'
import Home from './pages/Home'
import Log from './pages/Log'
import NotFound from './pages/NotFound'
import ScrollToTop from './components/ScrollToTop'
import Search from './pages/Search'
import SingleLogEntry from './pages/SingleLogEntry'
import SiteTitle from './components/SiteTitle'
import Todo from './pages/Todo'

import { userContext } from './contexts/userContext'

function AppRouter() {
  return (
    <Router>
      <ScrollToTop />
      <Switch>
        <userContext.Consumer>
          {({ user, updateUserState }) => 
          <>
            <SiteTitle />
            <Route path="/welcome" render={props => <GuestHome {...props} user={user} updateUserState={updateUserState} /> }/>
            <Route path="/" exact={true} render={props => <Home {...props} user={user} updateUserState={updateUserState} /> }/>
            <Route path="/settings" render={props => <Settings {...props} user={user} updateUserState={updateUserState} /> }/>
            <Route path="/add" render={props => <Add {...props} user={user} updateUserState={updateUserState} /> }/>
            <Route path="/log" exact render={props => <Log {...props} user={user} updateUserState={updateUserState} /> }/>
            <Route path="/log/:id/edit" exact render={props => <Edit {...props} user={user} updateUserState={updateUserState} /> }/>
            <Route path="/log/:slug" exact render={props => <SingleLogEntry {...props} user={user} updateUserState={updateUserState} /> }/>
            <Route path="/search" render={props => <Search {...props} user={user} updateUserState={updateUserState} /> }/>
            <Route path="/todo" render={props => <Todo {...props} user={user} updateUserState={updateUserState} /> }/>
            { user && user.cookies && user.cookies.length > 0 && <Nav /> }
            <Route path="*" component={NotFound} />
          </>
          }
        </userContext.Consumer>
      </Switch>
    </Router>
  )
}

export default AppRouter
