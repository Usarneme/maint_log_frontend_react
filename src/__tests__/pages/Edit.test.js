import React from 'react'
import Edit from '../../pages/Edit'
import renderer from 'react-test-renderer'

import AppRouter from '../../AppRouter'

describe('EDIT PAGE', () => {
  describe('\tNot Logged In', () => {
    it('redirects to / route', () => {
      const tree = renderer.create(
        <AppRouter>
          <Edit />
        </AppRouter>
        ).toJSON()
        // console.log(tree)
        // expect(tree[1].children[0].props.className).toEqual('guest__options')
        expect(tree).toMatchSnapshot()
    })
  })
})