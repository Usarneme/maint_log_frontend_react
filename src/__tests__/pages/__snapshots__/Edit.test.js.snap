// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EDIT PAGE 	Not Logged In redirects to / route 1`] = `
Array [
  <a
    className="siteTitle"
    href="/"
    onClick={[Function]}
  >
    <img
      alt="home"
      className="svg"
      description="home"
      src="home.svg"
    />
    <h1>
      Vehicle Maintenance Log
    </h1>
  </a>,
  <div
    className="inner"
  >
    <div
      className="guest__options"
    >
      <div
        className="guest__option login guest__option__active"
        onClick={[Function]}
      >
        <img
          alt="login"
          className="svg button"
          description="login icon"
          src="login.svg"
        />
        <h4>
          Login & View Your Maintenance Log
        </h4>
      </div>
      <div>
        <span>
          -or-
        </span>
      </div>
      <div
        className="guest__option register "
        onClick={[Function]}
      >
        <h4>
          Register & Start Your Maintenance Log
        </h4>
        <img
          alt="register"
          className="svg button"
          description="register icon"
          src="register.svg"
        />
      </div>
    </div>
    <div>
      <div
        className="inner"
      >
        <form
          className="card"
          method="POST"
          onSubmit={[Function]}
        >
          <h2>
            Login
          </h2>
          <label
            htmlFor="email"
          >
            Email Address
          </label>
          <input
            name="email"
            onChange={[Function]}
            placeholder="Enter email..."
            type="email"
            value=""
          />
          <label
            htmlFor="password"
          >
            Password
          </label>
          <input
            name="password"
            onChange={[Function]}
            placeholder="Enter password..."
            type="password"
            value=""
          />
          <input
            className="button"
            type="submit"
            value="Log In â†’"
          />
          <button
            className="button forgot__password__button"
            onClick={[Function]}
          >
            Forgot Your Password?
          </button>
        </form>
      </div>
    </div>
  </div>,
]
`;
